ultravox/training/helpers/prefetch_weights.py

```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
+from typing import List, Optional
+
+import huggingface_hub
+import transformers
+
+from ultravox.model import wandb_utils
+from ultravox.training import config_base
+
+ALLOW_PATTERNS = ["*.safetensors", "*.json"]
+
+
+def main(override_sys_args: Optional[List[str]] = None):
+    start = datetime.now()
+    print("Downloading weights ...")
+
+    args = config_base.get_train_args(override_sys_args)
+
+    for model_id in [args.text_model, args.audio_model]:
+        try:
+            # Download all model files that match ALLOW_PATTERNS
+            # This is faster than .from_pretrained due to parallel downloads
+            huggingface_hub.snapshot_download(
+                repo_id=model_id, allow_patterns=ALLOW_PATTERNS
+            )
+            # A backstop to make sure the model is fully downloaded even if ALLOW_PATTERNS is not enough
+            # Using `device_map="meta"` to avoid loading the weights into memory or device
+            transformers.AutoModel.from_pretrained(model_id, device_map="meta")
+        except huggingface_hub.utils.GatedRepoError as e:
+            raise e
+        except huggingface_hub.utils.RepositoryNotFoundError as e:
+            print(
+                f"Model {args.text_model} not found on HF Hub. Skipping download. Error: {e}"
+            )
```

Comment:
So it is ignored if the repository is not found?

Reply:
The assumption is that this is going to be a local model in this case.
I can add a check to see if the directory exists.

Reply:
Got it. Maybe the logic should be:

1. Fetch from HF
2. If repository not found on HF, check if local model exists
3. Fail with an error message

Reply:
I moved the `.from_pretrained` call outside of try/catch and updated comments. That should handle this case as well.